src/hb-blob.cc

```diff
@@ -545,53 +548,116 @@ hb_blob_create_from_file (const char *file_name)
   // Allison Lortie permission but changed to suit our need.
   bool writable = false;
   hb_memory_mode_t mm = HB_MEMORY_MODE_READONLY_MAY_MAKE_WRITABLE;
+  hb_mapped_file_t *file = (hb_mapped_file_t *) calloc (1, sizeof (hb_mapped_file_t));
+  if (unlikely (!file))
+  {
+    DEBUG_MSG (BLOB, nullptr, "Failed to allocate a local file handle");
+    return hb_blob_get_empty ();
+  }
 
+#ifdef HAVE_MMAP
   int fd = open (file_name, (writable ? O_RDWR : O_RDONLY) | _O_BINARY, 0);
-  if (unlikely (fd == -1)) return hb_blob_get_empty ();
-
-  hb_mapped_file_t *file = (hb_mapped_file_t *) calloc (1, sizeof (hb_mapped_file_t));
+  if (unlikely (fd == -1))
+  {
+    DEBUG_MSG (BLOB, nullptr, "Failed to open the file (mmap)");
```

Comment:
Can I pass something else the `DEBUG_MSG`? Or these debug messages are needed anyway?

Reply:
Not really needed.

Reply:
Gone, have a look